### ISSUE TITLE
vets-api: Remove eMIS references

### ISSUE THREAD

**Issue by rmtolmach** [department-of-veterans-affairs/va.gov-team#72200]
### Summary
To be done after #68316. 
As part of the code deletion for eMIS, we'll need to delete eMIS references in vets-api. According to my search, there are 931 references in 137 files. Split this work up into multiple PRs if possible so they're easier to review and there's less room for error. 

This includes, but it not limited to the emis section in [settings.yml](https://github.com/department-of-veterans-affairs/vets-api/blob/master/config/settings.yml) and any references to `Settings.emis`. Note: Any time an EMISRedis class is accessed, if the data is not already cached it calls through to [these service classes](https://github.com/department-of-veterans-affairs/vets-api/tree/master/lib/emis) to fetch from eMIS. We can delete all of those. 

### Tasks
- [x] Find all references to eMIS in vets-api repo
- [x] Logically break this work into more than one PR if possible. 
- [x] Delete references

### Acceptance criteria
- [x] All platform-owned references to eMIS in vets-api are gone
- [x] ArgoCD environments sync without errors after code is merged.

**Comment by jennb33** [comment]
Rewrite this ticket to follow 72611 and 72209 and use as remainder ticket for eMIS reference removals, per G&P meeting on 1/11/2024

**Comment by jennb33** [comment]
@rmtolmach picked this ticket up early.

**Comment by rmtolmach** [comment]
I have PR https://github.com/department-of-veterans-affairs/vets-api/pull/15320 up to delete the veteran_status code in vets-api. There will be more PRs.

**Comment by rmtolmach** [comment]
PR for deleting `pay` code in vets-api: https://github.com/department-of-veterans-affairs/vets-api/pull/15345

**Comment by rmtolmach** [comment]
- https://github.com/department-of-veterans-affairs/vets-api/pull/15320 merged but then was causing sign in problems in staging so it was reverted in https://github.com/department-of-veterans-affairs/vets-api/pull/15351. I will try again tomorrow, but _without_ the identity changes.
  - [ ] Investigate why it partially broke sign in.
- I merged https://github.com/department-of-veterans-affairs/vets-api/pull/15345 this evening.

**Comment by jennb33** [comment]
User string is expecting EMIS and Vet360 and @rmtolmach needs to retest it. In today's Stand, we discussed this and the FE's expectations. Might need to update the language (there is a lot of it). Might need to backlog a new ticket that would cover renaming instances of Vet360 to VA Profile (would need a new Epic that @jennb33 will spin up).

**Comment by rmtolmach** [comment]
This was still failing locally when I was testing. In addition to the changes in vets-api, I need to change [this line in vets-website](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/user/profile/utilities/index.js#L87). Once I had that in place, testing passed! I will put up a PR for that on Wednesday.

**Comment by rmtolmach** [comment]
I've been working on getting tests in my vets-website PR to pass https://github.com/department-of-veterans-affairs/vets-website/pull/27851 (and resolving warning messages)

**Comment by rmtolmach** [comment]
I created a new ticket to cover _only_ veteran status: https://github.com/department-of-veterans-affairs/va.gov-team/issues/75657. Moving all of that work there.

**Comment by rmtolmach** [comment]
PR to remove code related to Settings.emis: https://github.com/department-of-veterans-affairs/vets-api/pull/15584

**Comment by rmtolmach** [comment]
PR to delete the remaining `lib/emis` files and specs: https://github.com/department-of-veterans-affairs/vets-api/pull/15645

This is the largest chunk of work. I can see the light at the end of the tunnel after this PR is merged 🎉

**Comment by rmtolmach** [comment]
Small PR for removing some emis mentions: https://github.com/department-of-veterans-affairs/vets-api/pull/15658

**Comment by rmtolmach** [comment]
10-10 team to handle emis mentions in [app/services/form1010cg/service.rb](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/services/form1010cg/service.rb)

**Comment by rmtolmach** [comment]
Mobile backend engineers to handle references to [EMIS_STATUS502](https://github.com/search?q=repo%3Adepartment-of-veterans-affairs%2Fvets-api%20EMIS_STATUS502&type=code) in vets-api. [Ad Hoc Slack thread](https://adhoc.slack.com/archives/C0206QFMXFU/p1708958238362999).

**Comment by rmtolmach** [comment]
Final PR that will complete this ticket: https://github.com/department-of-veterans-affairs/vets-api/pull/15673

**Comment by rmtolmach** [comment]
All PRs merged and deployed. Closing.
